ws-nextjs-app/pages/[service]/send/[id]/FormContext/index.tsx

```diff
@@ -137,23 +143,28 @@ export const FormContextProvider = ({
       }
       formData.append(key, fieldValue as string);
     });
-
     try {
       const url = `https://www.bbc.com/ugc/send/${id}?said=${uuid()}`;
 
       const req = new XMLHttpRequest();
       req.responseType = 'json';
       req.open('POST', url, true);
 
+      req.upload.onloadstart = () => {
+        _setScreen('uploading');
+      };
+
       req.upload.onprogress = e => {
         setProgress(((e.loaded / e.total) * 100).toFixed(0));
       };
-
       req.onreadystatechange = () => {
         if (req.readyState === XMLHttpRequest.DONE) {
           setSubmitted(false);
           if (req.status === OK) {
             setSubmissionID(req.response.submissionId);
+            setTimeout(() => {
+              _setScreen('success');
+            }, 3000);
```

Comment:
Timeout logic is simple - it will always hold the page for 3 seconds + however long the upload takes.

Pro: Keeps the logic very simple

Con: If someone has to wait a while for their upload, they're always going to have to wait 3 seconds longer to find out if it's successful or errors

If the form server side errors - then everyone's going to be told that it's uploading for 3 seconds even if it's not uploading at all.

Reply:
I think this is fine, I like the approach here as I thought we'd need some state management in the Upload screen to handle this.